library_api/controllers/book_controller.go

```diff
@@ -0,0 +1,75 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"library_api/entities"
+	"library_api/repositories"
+	"net/http"
+	"sort"
+	"strconv"
+)
+
+func CreateBooks(w http.ResponseWriter, r *http.Request) { // TODO discuss about create action
+	name := r.FormValue("name")
+	author := r.FormValue("author")
+	year, err := strconv.Atoi(r.FormValue("year"))
+
+	if err != nil {
+		fmt.Println("year incorrect")
+		w.WriteHeader(http.StatusBadRequest)
+		panic(err)
+	}
+
+	if name == "" || author == "" {
+		fmt.Println("name or author incorrect")
+		w.WriteHeader(http.StatusBadRequest)
+	} else {
+		books := []entities.Book{
+			{name, author, year},
+		}
+
+		book_repository := repositories.NewBookRepository()
```

Comment:
it doesn't the best way to provide dependencies
Let's say we have a db connection under our book repo - in this case here we are going to create new connection for each time method gets called, it is pretty redundant
Lets create a separate struct for the Book controller entity and name in just Book (because we already know that it s controller because of package name) and add BookRepo param to it, so you will be able to provide it one time and use same connection in your controllers
Also don't forget to assign your controller's funcs as methods of Book controller struct